modules/standard/Regex.chpl

```diff
@@ -585,6 +585,18 @@ proc bytes.this(m:regexMatch) {
   else return b"";
 }
 
+
+module RegexDetail {
```

Comment:
Just curious, what's the motivation for moving this into its own module? Just so it's not polluting the namespace?

Reply:
Yeah but I think I commited the wrong thing, I had moved it inside `record regex {` to the same effect. Do you like that more

Reply:
I didn't even know we supported nested records. I think I'd avoid that. And FWIW given that it was no-doc'd before I didn't mind that it was in the namespace. Like it's technically there, but generally we reserve the right to use things starting with `chpl_` (and maybe `__`?), so I don't think we have to worry about namespace pollution for those cases. I'm fine leaving this as-is though if you prefer this.

Reply:
Yeah its a bit verbose but it does prevent the pollution. Going to leave